crates_io_tarball/src/lib.rs

```diff
@@ -89,25 +92,46 @@ pub fn process_tarball<R: Read>(
             ));
         }
 
-        if entry_path == vcs_info_path {
-            let mut contents = String::new();
-            entry.read_to_string(&mut contents)?;
-            vcs_info = CargoVcsInfo::from_contents(&contents).ok();
-        } else if entry_path == manifest_path || entry_path == manifest_path_lower {
-            // Try to extract and read the Cargo.toml from the tarball, silently
-            // erroring if it cannot be read.
-            let mut contents = String::new();
-            entry.read_to_string(&mut contents)?;
-            manifest = Some({
+        if entry_path.parent() == Some(pkg_root) {
```

Comment:
this should already be covered by the `if !entry_path.starts_with(pkg_name)` check above, right? admittedly with a more questionable condition though :D